examples/basic/basic.c

```diff
@@ -0,0 +1,117 @@
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include "umf/pools/pool_scalable.h"
+#include "umf/providers/provider_os_memory.h"
+
+#include <stdio.h>
+#include <string.h>
+
+int main(void) {
+    // A result object for storing UMF API result status
+    umf_result_t res;
+
+    // Create a memory provider
+    umf_memory_provider_ops_t *provider_ops = umfOsMemoryProviderOps();
+    umf_os_memory_provider_params_t params = umfOsMemoryProviderParamsDefault();
+    umf_memory_provider_handle_t provider;
+
+    res = umfMemoryProviderCreate(provider_ops, &params, &provider);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to create provider!");
+        return -1;
+    }
+    printf("OS Memory Provider created at %p\n", (void *)provider);
+
+    // Allocate memory from memory provider
+    size_t alloc_size = 5000;
+    size_t alignment = 0;
+    void *ptr_provider = NULL;
+
+    /// Get recommended page size for a proper alignment value
+    res = umfMemoryProviderGetRecommendedPageSize(provider, alloc_size,
+                                                  &alignment);
+    if (res != UMF_RESULT_SUCCESS || alignment == 0) {
+        printf("Failed to get recommended page size!");
+        goto memory_provider_destroy;
+    }
+    printf("Recommended page size is %zu bytes\n", alignment);
+
+    /// Allocate memory
+    res =
+        umfMemoryProviderAlloc(provider, alloc_size, alignment, &ptr_provider);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to allocate memory from the provider!");
+        goto memory_provider_destroy;
+    }
+    printf("Allocated memory at %p\n", ptr_provider);
+
+    /// Free allocated memory
+    res = umfMemoryProviderFree(provider, ptr_provider, alloc_size);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("Failed to free memory to the provider!");
+        goto memory_provider_destroy;
+    }
+    printf("Freed memory at %p\n", ptr_provider);
+
+    // Create a memory pool
+    umf_memory_pool_ops_t *pool_ops = umfScalablePoolOps();
+    void *pool_params = NULL;
+    umf_pool_create_flags_t flags = 0;
+    umf_memory_pool_handle_t pool;
+
+    res = umfPoolCreate(pool_ops, provider, pool_params, flags, &pool);
+    if (res != UMF_RESULT_SUCCESS) {
+        printf("\nFailed to create pool!");
+        goto memory_provider_destroy;
+    }
+    printf("\nScalable Memory Pool created at %p\n", (void *)pool);
+
+    // Allocate some memory
+    size_t num = 1;
+    alloc_size = 128;
+
+    char *ptr = umfPoolCalloc(pool, num, alloc_size);
+    if (!ptr) {
+        printf("Failed to allocate memory!");
+        goto memory_pool_destroy;
+    }
+
+    // Write a string to allocated memory
+    strcpy(ptr, "Allocated memory at");
+    printf("%s %p\n", ptr, (void *)ptr);
+
+    // Retrieve a memory pool from a pointer, available with memory tracking
+    // RFC: Remove this? With memory tracker disabled, NULL returned may be misleading
```

Comment:
Hm, this is a good point. Perhaps we should think about exposing some function like 'isMemoryTrackingEnabled' or something